src/GraphQL/TypeApi.hs

```diff
@@ -147,18 +152,22 @@ instance forall ks t m. (KnownSymbol ks, HasGraph m t, MonadThrow m, MonadIO m)
     let childResolver = buildResolver @m @t handler selectionSet
         name = toS (symbolVal (Proxy :: Proxy ks))
     in (name, childResolver)
-  buildFieldResolver _ f = ("", queryError ("buildFieldResolver got non AST.Field" <> show f <> ", query probably not normalized"))
+  buildFieldResolver _ f = ("x", queryError ("buildFieldResolver got non AST.Field" <> show f <> ", query probably not normalized"))
 
 
 instance forall ks t f m. (MonadThrow m, KnownSymbol ks, BuildFieldResolver m f, ReadValue t) => BuildFieldResolver m (Argument ks t :> f) where
   type FieldHandler m (Argument ks t :> f) = t -> FieldHandler m f
   buildFieldResolver handler selection@(AST.SelectionField (AST.Field _ _ arguments _ _)) =
     let argName = toS (symbolVal (Proxy :: Proxy ks))
         v = lookupValue argName arguments
-    in case v >>= readValue @t of
-      Left err -> ("", queryError err)
-      Right v' -> buildFieldResolver @m @f (handler v') selection
-  buildFieldResolver _ f = ("", queryError ("buildFieldResolver got non AST.Field" <> show f <> ", query probably not normalized"))
+    in case v of
+       Nothing -> case valueMissing @t argName of
+                     Left err' -> ("", queryError err')
+                     Right v' -> buildFieldResolver @m @f (handler v') selection
+       Just v' -> case readValue @t v' of
+                    Left err' -> ("", queryError err')
+                    Right v'' -> buildFieldResolver @m @f (handler v'') selection
+  buildFieldResolver _ f = ("y", queryError ("buildFieldResolver got non AST.Field" <> show f <> ", query probably not normalized"))
```

Comment:
You should probably remove the nested case by doing:

```haskell
v' = maybe (valueMissing @t argName) (readValue @t)
in case v' of
  Left err' -> ("", queryError ')
  Right v'' -> buildFieldResolver @m @f (handler v'') selection
```